Cannot build rust stdlib for VxWorks

It seems like building rust stdlib for vxworks on master is broken:

```bash

   Compiling cfg-if v1.0.1
   Compiling unwind v0.0.0 (/home/karthik/rust-m/library/unwind)
   Compiling rustc-std-workspace-alloc v1.99.0 (/home/karthik/rust-m/library/rustc-std-workspace-alloc)
   Compiling panic_unwind v0.0.0 (/home/karthik/rust-m/library/panic_unwind)
   Compiling gimli v0.32.0
   Compiling std_detect v0.1.5 (/home/karthik/rust-m/library/std_detect)
   Compiling miniz_oxide v0.8.9
   Compiling hashbrown v0.15.5
   Compiling addr2line v0.25.0
error[E0425]: cannot find value `AT_SYMLINK_NOFOLLOW` in crate `libc`
    --> library/std/src/sys/fs/unix.rs:2162:63
     |
2162 |             let flags = if follow_symlinks { 0 } else { libc::AT_SYMLINK_NOFOLLOW };
     |                                                               ^^^^^^^^^^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find function `utimensat` in crate `libc`
    --> library/std/src/sys/fs/unix.rs:2181:32
     |
2181 |             cvt(unsafe { libc::utimensat(libc::AT_FDCWD, p.as_ptr(), times.as_ptr(), flags) })?;
     |                                ^^^^^^^^^ help: a function with a similar name exists: `futimens`
     |
    ::: /home/karthik/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libc-0.2.177/src/vxworks/mod.rs:1273:5
     |
1273 |     pub fn futimens(fd: c_int, times: *const crate::timespec) -> c_int;
     |     ------------------------------------------------------------------- similarly named function `futimens` defined here

error[E0425]: cannot find value `AT_FDCWD` in crate `libc`
    --> library/std/src/sys/fs/unix.rs:2181:48
     |
2181 |             cvt(unsafe { libc::utimensat(libc::AT_FDCWD, p.as_ptr(), times.as_ptr(), flags) })?;
     |                                                ^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `O_NOFOLLOW` in crate `libc`
   --> library/std/src/sys/fs/mod.rs:136:36
    |
136 |         options.custom_flags(libc::O_NOFOLLOW);
    |                                    ^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `_SC_HOST_NAME_MAX` in crate `libc`
 --> library/std/src/sys/net/hostname/unix.rs:8:60
  |
8 |     let host_name_max = match unsafe { libc::sysconf(libc::_SC_HOST_NAME_MAX) } {
  |                                                            ^^^^^^^^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find function `strnlen` in crate `libc`
    --> library/std/src/sys/net/hostname/unix.rs:37:42
     |
  37 |                 let len = unsafe { libc::strnlen(buf.as_ptr().cast(), buf.capacity()) };
     |                                          ^^^^^^^ help: a function with a similar name exists: `strlen`
     |
    ::: /home/karthik/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libc-0.2.177/src/vxworks/mod.rs:1190:5
     |
1190 |     pub fn strlen(cs: *const c_char) -> size_t;
     |     ------------------------------------------- similarly named function `strlen` defined here
```

Most of these defines/functions are present in VxWorks but have not been added in libc which is causing this issue.

--- a/library/std/src/sys/fs/mod.rs
+++ b/library/std/src/sys/fs/mod.rs
@@ -122,7 +122,7 @@ pub fn set_permissions(path: &Path, perm: FilePermissions) -> io::Result<()> {
     with_native_path(path, &|path| imp::set_perm(path, perm.clone()))
 }
 
-#[cfg(unix)]
+#[cfg(all(unix, not(target_os = "vxworks")))]
 pub fn set_permissions_nofollow(path: &Path, perm: crate::fs::Permissions) -> io::Result<()> {
     use crate::fs::OpenOptions;
 
@@ -139,7 +139,7 @@ pub fn set_permissions_nofollow(path: &Path, perm: crate::fs::Permissions) -> io
     options.open(path)?.set_permissions(perm)
 }
 
-#[cfg(not(unix))]
+#[cfg(any(not(unix), target_os = "vxworks"))]
 pub fn set_permissions_nofollow(_path: &Path, _perm: crate::fs::Permissions) -> io::Result<()> {
     crate::unimplemented!(
         "`set_permissions_nofollow` is currently only implemented on Unix platforms"

